src/lib.rs

```diff
@@ -80,14 +80,29 @@ struct Reporter {
 #[pymethods]
 impl Reporter {
     #[new]
-    fn new() -> PyResult<Reporter> {
+    fn new(config: Option<&PyDict>) -> PyResult<Reporter> {
+        let mut agent_host_name: String = "127.0.0.1".to_string();
+        let mut agent_port: i32 = 6831;
+
+        if let Some(config) = config {
+            agent_host_name = config
+                .get_item("agent_host_name")
+                .unwrap()
+                .extract()
+                .unwrap_or("127.0.0.1".to_string());
+            agent_port = config
+                .get_item("agent_port")
+                .unwrap()
+                .extract()
+                .unwrap_or(6831);
```

Comment:
This will panic if `config` is set but doesn't include the given key, and will then default to `6831` if we fail to convert the py object returned by `get_item` into an int.

You probably want something like:

```suggestion
            if let Some(agent_port_arg) = config.get_item("agent_port") {
                agent_port = agent_port_arg.extract()?;
            }
```

which will return an error if it fails to convert the provided port as a string. Or if you want to provide a helpful exception:

```suggestion
            if let Some(agent_port_arg) = config.get_item("agent_port") {
                agent_port = agent_port_arg
                    .extract()
                    .map_err(|_| PyTypeError::new_err("'agent_port' must be an int"))?;
            }
```

Reply:
Thanks, this looks way more sane!

Really need to go through a Rust course to get a better handle on this. I guess the PyO3 bindings aren't helping either